### DISCUSSION TITLE
PPP Station 1 year time series (2020)

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#73] [Q&A] [answered]
### Description

Hello, 

we have used GROOPS for PPP determination for the ZIM 2 station. As the input files did not provide data consecutively for all 366 days in 2020 for (ZIM200CHE_R_2020XXX0000_01D_30S_MO.crx.gzr; XXX = 335 & 336 were not available) we ran the config files in 2 batches (001 -> 334 and 337 -> 366). Is there an option in GROOPS to ignore missing days / missing input files and continue the execution up to the last date in the loop interval definition 1 January 2020 to 31 December 2020?

thank you for your help
Best regards,
Jean Bemtgen

### Operating systems

- [ ] Linux
- [X] Windows
- [ ] macOS
- [ ] Other (please specify in the description)

**Comment by FFFFFG-IGG** [comment]
@Bemtgenj Hello, set continueAfterError "yes" of the loopProgram(contain the time series from 001~366) may solve your problem.

**Comment by Bemtgenj** [comment]
@FFFFFG-IGG TXS for the feedback - i will try that

**Comment by FFFFFG-IGG** [comment]
> @FFFFFG-IGG TXS for the feedback - i will try that
Excuse me, could you tell me if you have deploy ppp or network analyze with Beidou constellation?Since I'm not sure whether groops can process with Beidou, and I hadn't gotten any clue in the cookbook about that. Many thanks!
(may be developers can give me some illustration about that.) @sestras

**Comment by Waley04** [comment]
> > @FFFFFG-IGG TXS for the feedback - i will try that
> > Excuse me, could you tell me if you have deploy ppp or network analyze with Beidou constellation?Since I'm not sure whether groops can process with Beidou, and I hadn't gotten any clue in the cookbook about that. Many thanks!
> > (may be developers can give me some illustration about that.) @sestras

######################################
To deploy a PPP scenario for BeiDou constellation, you may try the following:

------- 01groopsConvert
In the loopPrn, set the path to "transmitterList.beidou.txt"
------- 02groopsGnssProcessing
In the transmitter, set the path to "transmitterList.beidou.txt"
Both the transmitter information and antennae definition (for the transmitter and receiver) should be set to igs14 i.e "transmitterInfo_igs14.{prn}.xml" and "antennaDefinition_igs14.dat"

Example 
Station = ALIC 
DOY = 305 in 2021.
Products = GBM 

------- Plots
![position_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173271222-2dd18cef-201d-4ae0-b9dc-99024e03ecf7.png)


![skyplot_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173269368-5d61d9aa-bc05-44a9-b706-9bf1320bec21.png)

######################################
#NB: Further details can be provided by the GROOPS Developers.

**Comment by FFFFFG-IGG** [comment]
> > > @FFFFFG-IGG TXS for the feedback - i will try that
> > > Excuse me, could you tell me if you have deploy ppp or network analyze with Beidou constellation?Since I'm not sure whether groops can process with Beidou, and I hadn't gotten any clue in the cookbook about that. Many thanks!
> > > (may be developers can give me some illustration about that.) @sestras
> 
> ###################################### To deploy a PPP scenario for BeiDou constellation, you may try the following:
> 
> ------- 01groopsConvert In the loopPrn, set the path to "transmitterList.beidou.txt" ------- 02groopsGnssProcessing In the transmitter, set the path to "transmitterList.beidou.txt" Both the transmitter information and antennae definition (for the transmitter and receiver) should be set to igs14 i.e "transmitterInfo_igs14.{prn}.xml" and "antennaDefinition_igs14.dat"
> 
> Example Station = ALIC DOY = 305 in 2021. Products = GBM
> 
> ------- Plots ![position_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173269117-adbea520-fdc7-4ef0-a814-2f717e72714d.png)
> 
> ![skyplot_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173269368-5d61d9aa-bc05-44a9-b706-9bf1320bec21.png)
> 
> ######################################

TXS for your kind reply! I'll try it.(If possible, could you give me your email address for further contact?)

**Comment by FFFFFG-IGG** [comment]
> > > @FFFFFG-IGG TXS for the feedback - i will try that
> > > Excuse me, could you tell me if you have deploy ppp or network analyze with Beidou constellation?Since I'm not sure whether groops can process with Beidou, and I hadn't gotten any clue in the cookbook about that. Many thanks!
> > > (may be developers can give me some illustration about that.) @sestras
> 
> ###################################### To deploy a PPP scenario for BeiDou constellation, you may try the following:
> 
> ------- 01groopsConvert In the loopPrn, set the path to "transmitterList.beidou.txt" ------- 02groopsGnssProcessing In the transmitter, set the path to "transmitterList.beidou.txt" Both the transmitter information and antennae definition (for the transmitter and receiver) should be set to igs14 i.e "transmitterInfo_igs14.{prn}.xml" and "antennaDefinition_igs14.dat"
> 
> Example Station = ALIC DOY = 305 in 2021. Products = GBM
> 
> ------- Plots ![position_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173271222-2dd18cef-201d-4ae0-b9dc-99024e03ecf7.png)
> 
> ![skyplot_2021-11-01 alic](https://user-images.githubusercontent.com/101952777/173269368-5d61d9aa-bc05-44a9-b706-9bf1320bec21.png)
> 
> ###################################### #NB: Further details can be provided by the GROOPS Developers.

Excuse me, I've tried the schema you provided, but get totally different results. (Station = ALIC DOY = 305 in 2021. Products = GBM),ppp results of fix(figure 1) and float(figure 2) ambiguity are list bellow:
![alic_305](https://user-images.githubusercontent.com/59271400/178101037-08c59855-4c46-4f0a-a7ec-2fd1599c7605.png)
figrue 1
![alic_305](https://user-images.githubusercontent.com/59271400/178101162-64d1f231-328f-49c7-ae6a-091411f7d317.png)
figure 2

Since PPP using GPS and GALILEO system are normal, I do not know what goes wrong with ppp using BEIDOU system(Using GLONASS also met this problem) .
Here I would like to ask some details about your schema:
1.Which version of groops are you using?(My groops is downloaded in 2022-3-7)
2.Do you only use BEIDOU system in your schema?
3.Do you use all satellites of BEIDOU system?
4.Which band of carrier-phase rang do you use?
Many thanks, and I'm looking forward to your reply as soon as possible.

**Comment by Waley04** [comment, marked as answer]
Greetings,

1. Which version of groops are you using?

Here is the version I am using:
![dateCompiles](https://user-images.githubusercontent.com/101952777/178152488-c62c8e3b-6962-4879-9584-63b55fd016c3.PNG)
Note that I also have the groops version compiled in January 2022 and another version compiled in November 2021. The example works on all versions I have.

2. Do you only use BEIDOU system in your schema?

Yes, I use the BDS constellation only. 

3. Do you use all satellites of BEIDOU system?

Yes, all the satellites with useful observations are used. In this example, 37 PRNs are used (PRN C06 to C46). 
You may refer to this usedBEIDOU_PRNs_2021-11-01.txt
[usedBEIDOU_PRNs_2021-11-01.txt](https://github.com/groops-devs/groops/files/9079577/usedBEIDOU_PRNs_2021-11-01.txt)

4. Which band of carrier-phase rang do you use?

For this example, I used B1 and B3 in-phase signals. 

I hope this answers my example. 

Best wishes,

Waley.

###########################################################
Disclaimer: I just know how to deploy the BDS PPP test in this software. 
GROOPS developers can explain the quality of the solution.
###########################################################

**Comment by FFFFFG-IGG** [comment]
> For this example, I used B1 and B3 in-phase signals.

It's so weird! There is still something wrong with my ppp results with Beidou, eventhough I use the same products you've mentioned. If possible, could you please provide your config.xml and GBM products, obs files? It may be of great help to find out what's going wrong with my strategy. 
Many thanks!
